encodings/fsst/src/array.rs

```diff
@@ -42,6 +45,21 @@ pub struct FSSTArray {
     /// Lengths of the original values before compression, can be compressed.
     uncompressed_lengths: ArrayRef,
     stats_set: ArrayStats,
+
+    /// Memoized compressor used for push-down of compute by compressing the RHS.
+    compressor: Arc<LazyLock<Compressor, Box<dyn Fn() -> Compressor + Send>>>,
```

Comment:
interesting, why does it need the `Send` hint? Compressor is a struct, shouldn't it be able to infer that it's send?

Reply:
Not sure, it complains that the future isn't send, therefore the arc isn't sync